crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -1899,7 +1995,55 @@ fn attach_name_to_expr(ident: Ident, expr: Expr, extra_items: &mut Vec<ModuleIte
         }],
         ..Default::default()
     })))));
+}
+
+fn attach_name_to_default_expr(ident: &Ident, extra_items: &mut Vec<ModuleItem>) {
+    // Assign a name with `Object.defineProperty($$ACTION_0, 'name', {value: 'default'})`
+    extra_items.push(ModuleItem::Stmt(Stmt::Expr(ExprStmt {
+        span: DUMMY_SP,
+        expr: Box::new(Expr::Call(CallExpr {
+            span: DUMMY_SP,
+            callee: Callee::Expr(Box::new(Expr::Member(MemberExpr {
+                span: DUMMY_SP,
+                obj: Box::new(Expr::Ident(Ident::new(
+                    format!("Object").into(),
+                    DUMMY_SP,
+                    ident.ctxt,
+                ))),
+                prop: MemberProp::Ident(IdentName::new(format!("defineProperty").into(), DUMMY_SP)),
+            }))),
+            args: vec![
+                ExprOrSpread {
+                    spread: None,
+                    expr: Box::new(Expr::Ident(ident.clone())),
+                },
+                ExprOrSpread {
+                    spread: None,
+                    expr: Box::new(Expr::Lit(Lit::Str("name".into()))),
+                },
+                ExprOrSpread {
+                    spread: None,
+                    expr: Box::new(Expr::Object(ObjectLit {
+                        span: DUMMY_SP,
+                        props: vec![
+                            PropOrSpread::Prop(Box::new(Prop::KeyValue(KeyValueProp {
+                                key: PropName::Str("value".into()),
+                                value: Box::new(Expr::Lit(Lit::Str("default".into()))),
+                            }))),
+                            PropOrSpread::Prop(Box::new(Prop::KeyValue(KeyValueProp {
+                                key: PropName::Str("writable".into()),
+                                value: Box::new(Expr::Lit(Lit::Bool(false.into()))),
+                            }))),
+                        ],
+                    })),
+                },
```

Comment:
Haven't verified this but IIRC lit can be directly converted:

```suggestion
                ExprOrSpread {
                    spread: None,
                    expr: Box::new("name".into()),
                },
                ExprOrSpread {
                    spread: None,
                    expr: Box::new(Expr::Object(ObjectLit {
                        span: DUMMY_SP,
                        props: vec![
                            PropOrSpread::Prop(Box::new(Prop::KeyValue(KeyValueProp {
                                key: PropName::Str("value".into()),
                                value: Box::new("default".into()),
                            }))),
                            PropOrSpread::Prop(Box::new(Prop::KeyValue(KeyValueProp {
                                key: PropName::Str("writable".into()),
                                value: Box::new(false.into()),
                            }))),
                        ],
                    })),
                },
```

Reply:
Yup, that works, thank you!